descqa/NumberDensityVersusRedshift.py

```diff
@@ -0,0 +1,322 @@
+from __future__ import unicode_literals, absolute_import 
+import os
+import numpy as np
+import re
+import math
+#from GCR import GCRQuery
+from .base import BaseValidationTest, TestResult
+from .plotting import plt
+
+__all__ = ['NumberDensityVersusRedshift']
+
+#observations
+Coil2004='Coil2004'
+Coil2004_magbin='Coil2004_magbin'
+Coil2004_maglim='Coil2004_maglim'
+DEEP2_JAN='DEEP2_JAN'
+Coil2004_lbl='Coil et. al. 2004'
+DEEP2_JAN_lbl='DEEP2 (JAN, p.c.)'
+
+#plotting constants
+figx_p=9
+figy_p=11
+lw2=2
+fsize=16
+lsize=10  
+default_colors=['blue','r','m','g','navy','y','purple','gray','c','orange','violet','coral','gold','orchid','maroon','tomato','sienna','chartreuse','firebrick','SteelBlue']
+validation_color='black'
+
+class NumberDensityVersusRedshift(BaseValidationTest):
+    """
+    validation test to show N(z) distributions
+    """
+    def __init__(self,z='redshift_true',band='i',N_zbins=44,zlo=0.,zhi=1.1,observation='',datafile='',m_lo=27,m_hi=18,ncolumns=2,normed=True, **kwargs):
+        
+        #catalog quantities
+        self.z = z
+        possible_mag_fields = ('mag_{}_lsst',
+                               'mag_{}_sdss',
+                               'mag_{}_des',
+                              )
+        self.possible_mag_fields = [f.format(band) for f in possible_mag_fields]
+        self.band = band
+        
+        #z-bounds and binning
+        self.zlo = zlo
+        self.zhi = zhi
+        self.N_zbins = N_zbins
+        self.zbins = np.linspace(zlo,zhi,N_zbins+1)
+        self.filters = [(lambda z: (z > zlo) & (z < zhi), self.z)]
+
+        #validation data
+        self.validation_data={}
+        possible_observations = ['Coil2004_maglim','Coil2004_magbin','DEEP2_JAN']
+        self.observation = observation
+        #check for valid combinations
+        if(len(observation)==0 and len(datafile)==0):
```

Comment:
no parentheses for if, please edit all similar instances

Reply:
Done